packages/flutter_tools/test/commands.shard/hermetic/assemble_test.dart

```diff
@@ -13,108 +13,98 @@ import 'package:flutter_tools/src/build_system/build_system.dart';
 import 'package:flutter_tools/src/cache.dart';
 import 'package:flutter_tools/src/commands/assemble.dart';
 import 'package:flutter_tools/src/convert.dart';
-import 'package:mockito/mockito.dart';
 import 'package:flutter_tools/src/globals.dart' as globals;
 
 import '../../src/common.dart';
 import '../../src/context.dart';
+import '../../src/fakes.dart';
 import '../../src/testbed.dart';
 
 void main() {
   Cache.disableLocking();
   Cache.flutterRoot = '';
+  final StackTrace stackTrace = StackTrace.current;
 
-  final Testbed testbed = Testbed(overrides: <Type, Generator>{
-    BuildSystem: ()  => MockBuildSystem(),
-    Cache: () => FakeCache(),
-  });
-
-  testbed.test('flutter assemble can run a build', () async {
-    when(globals.buildSystem.build(any, any, buildSystemConfig: anyNamed('buildSystemConfig')))
-      .thenAnswer((Invocation invocation) async {
-        return BuildResult(success: true);
-      });
+  testUsingContext('flutter assemble can run a build', () async {
     final CommandRunner<void> commandRunner = createTestCommandRunner(AssembleCommand());
     await commandRunner.run(<String>['assemble', '-o Output', 'debug_macos_bundle_flutter_assets']);
 
     expect(testLogger.traceText, contains('build succeeded.'));
+  }, overrides: <Type, Generator>{
+    BuildSystem: ()  => TestBuildSystem.all(BuildResult(success: true)),
+    Cache: () => FakeCache(),
+    FileSystem: () => MemoryFileSystem.test(),
+    ProcessManager: () => FakeProcessManager.any(),
   });
 
-  testbed.test('flutter assemble can parse defines whose values contain =', () async {
-    when(globals.buildSystem.build(any, any, buildSystemConfig: anyNamed('buildSystemConfig')))
-      .thenAnswer((Invocation invocation) async {
-        expect((invocation.positionalArguments[1] as Environment).defines, containsPair('FooBar', 'fizz=2'));
-        return BuildResult(success: true);
-      });
+  testUsingContext('flutter assemble can parse defines whose values contain =', () async {
     final CommandRunner<void> commandRunner = createTestCommandRunner(AssembleCommand());
     await commandRunner.run(<String>['assemble', '-o Output', '-dFooBar=fizz=2', 'debug_macos_bundle_flutter_assets']);
 
     expect(testLogger.traceText, contains('build succeeded.'));
+  }, overrides: <Type, Generator>{
+    BuildSystem: ()  => TestBuildSystem.all(BuildResult(success: true), (Target target, Environment environment) {
+      expect(environment.defines, containsPair('FooBar', 'fizz=2'));
```

Comment:
Any chance this could be refactored to check the received environment, exception, perf measurement, etc inside the test instead of the override?

Reply:
They'd still need to be testUsingContext due to the command class usage

Reply:
oh sorry I misunderstood. Yeah good point, will take a look at doing that